beetsplug/kodinfo.py

```diff
@@ -0,0 +1,448 @@
+                data[3],
+                data2["strStyle"] or '',
+                data2["strMood"] or '',
+                data2["intBornYear"] or '',
+                data2["intFormedYear"] or '',
+                data2["strBiographyEN"] or '',
+                data2["intDiedYear"] or '',
+                data2["strDisbanded"] or '',
+                data2["strArtistThumb"] or '',
+                data2["strArtistFanart"] or ''))
+        return list(out_data.split(';'))
+
+
+def artist_albums(artistid):
+    """Get artist's albums from beets library."""
+    albumdata = []
+    for album in lib.albums(artistid):
+        row = album.album, album.original_year
+        albumdata.append(list(tuple([row[1], row[0]])))  # create sortable list
+    # sort list to start with first release/album
+    albumlist = (sorted(albumdata))
+    return albumlist
+
+
+def album_info(albumid):
+    """Collect album information from beets lib and audiodb.com."""
+    for album in lib.albums(albumid):
+        data = (
+            album.albumartist,
+            album.mb_albumartistid,
+            album.mb_releasegroupid,
+            album.album,
+            album.genre,
+            album.comp,
+            album.label,
+            album.albumtype,
+            album.mb_albumid)
+        date = album.original_year, album.original_month, album.original_day
+        rel_date = (
+            "%s-%s-%s" %
+            (date[0], format(
+                date[1], '02'), format(
+                date[2], '02')))
+        url = audiodb_url + "{0}/album-mb.php?i=".format(
+            audio_db_key)
+
+        if data[5] == 0:
+            comp = 'False'
+        else:
+            comp = 'True'
+
+        try:
+            response = urllib.request.urlopen(url + data[2])
+            data2 = json.load(response)["album"][0]
+
+        except (ValueError, TypeError):
+            # catch simplejson.decoder.JSONDecodeError and load emptydata
+            data2 = json.loads(emptyalbum)["album"][0]
+
+        out_data = ("{0};{1};{2};{3};{4};{5};{6};{7};{8};{9};{10};{11};{12};"
+                    "{13};{14}".format((data[3]),
+                                       (data[8]),
+                                       (data[0]),
+                                       (data[4]),
+                                       (data2["strStyle"]) or '',
+                                       (data2["strMood"]) or '',
+                                       (data2["strTheme"]) or '',
+                                       (comp),
+                                       (data2["strReview"]) or '',
+                                       (data[7]),
+                                       (rel_date),
+                                       (data[6]),
+                                       (data2["intScore"]) or '',
+                                       (date[0]),
+                                       (data2["strAlbumThumb"]) or ''))
+        return list(out_data.split(';'))
```

Comment:
Formatting a string with semicolons and then splitting again on semicolons seems like it could cause trouble… is there any particular reason why this shouldn't just return a list? (Or, better yet, a dictionary with descriptive names for each field?)